### PR TITLE
Feature/vendor location

### PR DESCRIPTION
**Pull request by Pma913**
____ Wrote Tests __x__ Implemented ____ Reviewed

## Neccesary checkmarks
- [ ] All Tests are Passing in all environments
- [x] The code will run locally
## Type of change
- [x] New feature
- [ ] Bug Fix
- [x] Refactor
## Description of Change:

  description: The main changes that this PR will implement are utilizing websockets to broadcast food truck address change and to add functionality to the notifications bell. There are some minor code cleanups/refactors added in.
    
## Requested feedback: 
    
  I'd like feedback on... anything is fair game. I do have some concern on the session storage that I implemented for maintaining the focus on vendor/customers button. The main thing is that I think it could use some refactoring, but it does seem to be working without bugs. Any feedback on that would be appreciated

## Check the correct boxes

- [x] This broke nothing
- [ ] This broke some stuff
- [ ] This broke everything

## Testing Changes

- [x] No Tests have been changed
- [ ] Some Tests have been changed
- [ ] All of the Tests have been changed(Please describe what in the world happened)

## Checklist:

- [x] My code has no unused/commented out code
- [ ] My code has no binding.pry's
- [x] I have reviewed my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have fully tested my code

(For Fun!)Please Include a link to a gif of your feelings about this branch

Link:

### CODE REVIEW

File: src/Components/App/App.js [hunk 1]
```diff
@@ -82,28 +88,29 @@ const App = () => {
...
-      useEffect(() => {
-        fetchData()
-  }
             
   useEffect(() => {
     fetchData()
 
     socket.on('receive_data', (data) => {
       setVendors([data.truck, ...data.updatedVendors])
+      setPushNote([...pushNote, { vendorName: data.truck.name }])
```
Comment (emlindvall) [inline src/Components/App/App.js:97]: SO excited to see this in practice. Thanks for doing all the hard work to get this implemented!

File: src/Components/App/App.js [hunk 2]
```diff
@@ -115,7 +122,7 @@ const App = () => {
       />
       <Route path="/vendor-view" element={
         <div>
-          <VendorView toggleLive={sendData} vendor={vendors.find(v => v.id == 1)} currentVendor={currentVendor}/>
+          <VendorView toggleLive={sendData} getAddress={sendAddress} vendor={vendors.find(v => v.id == 9)} currentVendor={currentVendor}/>
```
Comment (emlindvall) [inline src/Components/App/App.js:125]: You'll have to walk me through changing this from an id of 1 to an id of 9-- not sure I get it just looking at the code. Thanks in advance!

File: src/Components/Header/Header.js [hunk 3]
```diff
@@ -24,10 +24,18 @@ const Header = ({ togView, currentUser }) => {
   
   const selUser = JSON.parse(sessionStorage.getItem('SEuserBtnStatus')) || 'customer'
 
+  const notification = () => {
+    if (notifs.length) {
+      return `${notifs[0].vendorName} has gone live`
```
Comment (emlindvall) [inline src/Components/Header/Header.js:29]: sooooo cool

File: src/Components/Status/Status.css [hunk 4]
```diff
@@ -144,6 +144,29 @@
...
+  border: none;
+  border-radius: 25px;
+  cursor: pointer;
+  transition: all .3s ease-in-out;
+  margin: 10px;
+}
+
+.broadcast-btn:hover {
+  width: 155px;
+
```
Comment (emlindvall) [inline src/Components/Status/Status.css:161]: Random extra space, here

**Review by emlindvall — COMMENTED** [review]
Phenomenal work, Patrick!

**Review by emlindvall — APPROVED** [review]

File: src/Components/App/App.js (hunk 2 above)
Comment (Pma913) [reply src/Components/App/App.js:125]: There is nothing special there. It was to change which  vendor is hardcoded to the vendor view. This way it is Taj Palace, the one that by default is rendered at the top of the list on the main page.

File: src/Components/App/App.js [hunk 5]
```diff
@@ -37,6 +38,11 @@ const App = () => {
     socket.emit("send_data", { updatedVendors: vendors.filter(v => v.id !== truck.id), truck: truck });
```
Comment (JacobMacFarlane) [inline src/Components/App/App.js:38]: so are you sedning data to all vendors whos id dont match up?

File: src/Components/Header/Header.js [hunk 6]
```diff
@@ -24,10 +24,18 @@ const Header = ({ togView, currentUser }) => {
   
   const selUser = JSON.parse(sessionStorage.getItem('SEuserBtnStatus')) || 'customer'
```
Comment (JacobMacFarlane) [inline src/Components/Header/Header.js:25]: great work around Is SEuserBtnStatus a session storage item or something you created?

**Review by JacobMacFarlane — COMMENTED** [review]
Looks really good, would love to see it in action but also your thought process of how you got to this point with websockets since im still having a hard time trying to grasp it

**Pull request merged**